ucte/ucte-converter/src/main/java/com/powsybl/ucte/converter/UcteExporter.java

```diff
@@ -0,0 +1,1072 @@
+            );
+        } while (iidmIdToUcteNodeCodeId.values().contains(ucteNodeCode));
+        iidmIdToUcteNodeCodeId.put(id, ucteNodeCode);
+    }
+
+    /**
+     * create a unique UcteElementId with the given information in parameters
+     *
+     */
+    UcteElementId convertUcteElementId(UcteNodeCode ucteNodeCode1, UcteNodeCode ucteNodeCode2, String id, Terminal terminal1, Terminal terminal2) {
+        if (isUcteNodeId(terminal1.getBusBreakerView().getConnectableBus().getId()) &&
+                isUcteNodeId(terminal2.getBusBreakerView().getConnectableBus().getId())) {
+            return new UcteElementId(
+                    ucteNodeCode1,
+                    ucteNodeCode2,
+                    id.charAt(18));
+        } else {
+            return generateUcteElementId(id, ucteNodeCode1, ucteNodeCode2, null);
+        }
+    }
+
+    /**
+     * increment the last character of the id from 'a' to 'z' until the id is unique, and return it
+     *
+     * @param id the id of the original iidm component to convert
+     * @param ucteNodeCode1 the ucteNodeCode used to create the UcteElementId
+     * @param ucteNodeCode2 the ucteNodeCode used to create the UcteElementId
+     * @return the generated UcteElementId
+     */
+    UcteElementId generateUcteElementId(String id, UcteNodeCode ucteNodeCode1, UcteNodeCode ucteNodeCode2, Character propertyOrderCode) {
+        if (iidmIdToUcteElementId.containsKey(id)) {
+            return iidmIdToUcteElementId.get(id);
+        } else {
+            int orderCode = 'a';
+            UcteElementId ucteElementId = null;
+
+            if (propertyOrderCode != null) { // First we try to generate an ucteElementId with the orderCode in the property
+                ucteElementId = new UcteElementId(ucteNodeCode1, ucteNodeCode2, propertyOrderCode);
+                if (!iidmIdToUcteElementId.values().contains(ucteElementId)) {
+                    iidmIdToUcteElementId.put(id, ucteElementId);
+                    return ucteElementId;
+                }
+            }
+
+            do {
+                if (orderCode <= 'z') {
+                    ucteElementId = new UcteElementId(ucteNodeCode1, ucteNodeCode2, (char) orderCode);
+                } else {
+                    LOGGER.warn("There is more than 26 element with the the id starting by '{}'", ucteElementId);
+                    return null;
+                }
+                orderCode++;
+            } while (iidmIdToUcteElementId.values().contains(ucteElementId));
+            iidmIdToUcteElementId.put(id, ucteElementId);
+            return ucteElementId;
+        }
+    }
+
+    /**
+     * check if the id of the line match with the id of a tie line in ucte
+     *
+     * @param line is a tie line ?
+     * @return true if the parameter is a tie line
+     */
+    boolean isUcteTieLineId(Line line) {
+
+        MergedXnode mergedXnode = line.getExtension(MergedXnode.class);
+
+        return mergedXnode != null
```

Comment:
If line.isTileLine() is `true`, should this line be considered as a TieLine?

Reply:
This method is only to check if the line has the needed extension to recreate the ucte TieLine